.github/workflows/CI.yml

```diff
@@ -111,58 +99,41 @@ jobs:
           echo "${OE_LICENSE_TEXT}" > ${OE_LICENSE}
 
       - name: Install package
-        shell: bash -l {0}
         run: |
           conda remove --force openff-toolkit openff-toolkit-base
           python setup.py develop --no-deps
 
-      - name: Check installed toolkits
-        shell: bash -l {0}
+      - name: Remove undesired toolkits
         run: |
-          if [[ "$OPENEYE" == true ]]; then
-            python -c "from openff.toolkit.utils.toolkits import OPENEYE_AVAILABLE; assert OPENEYE_AVAILABLE, 'OpenEye unavailable'"
-          fi
-          if [[ "$OPENEYE" == false ]]; then
-            if [[ $(conda list | grep openeye-toolkits) ]]; then
-              conda remove --force openeye-toolkits --yes
-            fi
-            python -c "from openff.toolkit.utils.toolkits import OPENEYE_AVAILABLE; assert not OPENEYE_AVAILABLE, 'OpenEye unexpectedly found'"
-          fi
-          if [[ "$RDKIT" == true ]]; then
-            python -c "from openff.toolkit.utils.toolkits import RDKIT_AVAILABLE; assert RDKIT_AVAILABLE, 'RDKit unavailable'"
-          fi
-          if [[ "$RDKIT" == false ]]; then
-            if [[ $(conda list | grep rdkit) ]]; then
-              conda remove --force rdkit --yes
-            fi
-            if [[ $(conda list | grep ambertools) ]]; then
-              conda remove --force ambertools --yes
-            fi
-            python -c "from openff.toolkit.utils.toolkits import RDKIT_AVAILABLE; assert not RDKIT_AVAILABLE, 'RDKit unexpectedly found'"
+          if [[ ${{ env.PACKAGES_TO_REMOVE }} ]]; then
+            for cpkg in ${{ env.PACKAGES_TO_REMOVE }}; do
+              echo $cpkg
+              if [[ $(conda list | grep $cpkg) ]]; then conda remove --force $cpkg --yes ; fi
+            done
           fi
 
+      - name: Check installed toolkits
+        run: |
+          for tk in ${{ env.TOOLKIT_CHECKS }}; do
+            python -c "from openff.toolkit.utils.toolkits import ${tk}_AVAILABLE; assert ${tk}_AVAILABLE, '${tk} unavailable'"
+          done
+
       - name: Environment Information
-        shell: bash -l {0}
         run: |
           conda info
           conda list
 
       - name: Check links
-        shell: bash -l {0}
-        if: ${{ matrix.cfg.rdkit == 'true' && matrix.cfg.openeye == 'TRUE' }}
-        run: |
-          pytest -r fE --tb=short openff/toolkit/tests/test_links.py
+        if: ${{ matrix.rdkit == true && matrix.openeye == true }}
+        run: pytest -r fE --tb=short openff/toolkit/tests/test_links.py
 
       - name: Run mypy
-        shell: bash -l {0}
         # Try to prevent failure from preventing later steps, but also display red X
         # https://stackoverflow.com/a/62112985/4248961
```

Comment:
This comment (which I think I wrote) seems to be wrong with the current code. Might as well remove it now.

Reply:
Is it still desired? Could you use `if: always()` to always run something, or `continue-on-error`?

Reply:
The thing I worry about, and the reason why it's here at all, is when a dumb little thing breaks `mypy` and therefore prevents unit tests from running, losing the ~5 mins of previous investment in the run getting conda and everything else set up. Probably the best way to do this, though, is just move this step to the very end?